dedup.py

```diff
@@ -0,0 +1,480 @@
+- 1: Round to nearest 0.1s (1.73s → 1.7s)
+"""
+
+# =============================================================================
+# CONFIGURATION SUMMARY
+# =============================================================================
+"""
+Current deduplication configuration:
+├─ Similarity Detection: 0.8 threshold (strict)
+├─ Text Fingerprinting: 5-character n-grams  
+├─ LSH Performance: 16 bands × 128 rows = 2048 permutations
+├─ Score Grouping: 4 decimal places for leaderboard scores
+└─ Duration Grouping: 0 decimal places for execution times
+
+To adjust deduplication sensitivity:
+- Increase FUZZY_SIMILARITY_THRESHOLD (0.8→0.9) for stricter deduplication
+- Decrease FUZZY_SIMILARITY_THRESHOLD (0.8→0.7) for more aggressive deduplication  
+- Adjust NGRAM_SIZE for different text lengths (3-4 for short, 5-7 for long)
+"""
+
+def remove_duplicates(data_dict: Dict[str, Dict[bool, Dict[Union[float, int], List[Dict]]]]):
+    """
+    Remove exact duplicates from the nested data structure returned by get_sorted_hf_data.
+    
+    Args:
+        data_dict: Nested dictionary structure from get_sorted_hf_data
+        
+    Returns:
+        Dictionary with same structure but duplicates removed
+    """
+    deduplicated_dict = {}
+    
+    for run_mode, score_duration_dict in tqdm.tqdm(data_dict.items(), desc="Processing run modes"):
+        deduplicated_dict[run_mode] = {}
+
+        for run_success, run_success_dict in tqdm.tqdm(score_duration_dict.items(), desc=f"Processing {run_mode}", leave=False):
+            deduplicated_dict[run_mode][run_success] = {}
+            for score_duration, rows in tqdm.tqdm(run_success_dict.items(), desc=f"Processing {run_mode}", leave=False):
+                # Use a dictionary to track unique entries by their content hash
+                unique_entries = {}
+                
+                for row in tqdm.tqdm(rows, desc=f"Processing {run_mode} {score_duration}", leave=False):
+                    # Create a hash of the relevant content (assuming 'input' or similar field exists)
+                    # If the row has an 'input' field, use that; otherwise use the entire row
+                    content = row.get('code', "")
+                    content_hash = hashlib.sha256(content.encode()).hexdigest()
+                    
+                    if content_hash not in unique_entries:
+                        unique_entries[content_hash] = row
+                    else:
+                        # If duplicate found, keep the one with better metrics
+                        existing_row = unique_entries[content_hash]
+                        
+                        # For leaderboard mode with successful runs, prefer higher scores
+                        if run_mode == 'leaderboard' and row.get('run_passed') == True:
+                            if row.get('run_score', 0) > existing_row.get('run_score', 0):
+                                unique_entries[content_hash] = row
```

Comment:
I think scores are still lower = better; run.duration is the end-to-end wallclock time for the entire run, including, e.g., testing code, whereas score is the geomean of all benchmarks